src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -278,4 +285,37 @@ private boolean isConnectedToPedestrianNavigableHighway(final Edge edge)
     {
         return edge.connectedEdges().stream().anyMatch(HighwayTag::isPedestrianNavigableHighway);
     }
+
+    /**
+     * Checks if the edge is within or intersects airport polygon. Airport polygon is an atlas area
+     * with {@link AerowayTag} tag.
+     *
+     * @param edge
+     *            any edge
+     * @return true if the edge is within or intersects airport polygon
+     */
+    private boolean intersectsAirportOrBuilding(final Edge edge)
+    {
+        return StreamSupport.stream(edge.getAtlas()
+                .areasIntersecting(edge.bounds(),
+                        area -> Validators.hasValuesFor(area, AmenityTag.class)
+                                || BuildingTag.isBuilding(area)
+                                || Validators.hasValuesFor(area, AerowayTag.class))
+                .spliterator(), false).count() > 0;
+    }
+
+    /**
+     * Checks if the edge is car navigable or not. A car navigable edge is an edge with highway type
+     * within the car navigable enum set and with certain accessibility tag value = !no
+     *
+     * @param edge
+     *            any edge
+     * @return true if the edge is car navigable, else false
+     */
+    private boolean isCarNavigableHighway(final Edge edge)
```

Comment:
Have you considered moving this to a location (`TypePredicates`) where other checks can pick this up?
Also, is motor_vehicle=yes required to use motor_vehicle=no to determine navigability ?

Reply:
Having this method as a utility function in TypePredicate will be very useful. I did not go for it this time because it requires some deep dive into the definition of navigability. In case of SinkIsland, @Keahi and I decided to go with "!motor_vehicle=no" because of the amount of flags that would be introduced by "!motor_vehicle=yes" and also because the requirement specifically cited FPs with "motorvehicle=no" tags. But ideally, "motorvehicle=yes" should determine the navigability of an edge.

Reply:
@danielduhh I have updated to use a more generic tag predicate in: https://github.com/sayas01/atlas-checks/pull/12/commits/24afba08d193c639b3f7e7385a3befaf9c0bca09

Reply:
There are definite intricacies to navigability in OSM. In this case it looks like the largest piece you are missing is motor_vehicle/motorcar/vehicle=yes overriding access=no. I think it is definitely worth researching and writing out a utility function for this.

Reply:
@kkonishi2 what do you think?

Reply:
It seems like there's a need for a unified definition of OSM navigability in this repo (or in atlas itself). 

The research involved in doing that properly might be outside the scope of this correction, but I would vote that we look into doing the necessary research. We could either say that this PR is blocked until that change happens, or we could push this through and refactor it to use the navigability definition later. Either way is fine with me -- what do you all think?

Reply:
check************ @nachtm

Reply:
I think that the way we have it set up is fine. Simply looking for a motorway=yes tag may lead to unforeseen issues. For example, you're able to tag a Way as access=private and motor_vehicle=yes, potentially leading to some confusing situations. Were this an internal check, I would be more concerned with trying to keep standards uniform, but because this is external, I think it's pretty safe to treat Ways with tags such as access=customer as essentially unnavigable ways. Most routing engines will assign high costs to roads based on these tags, so these are essentially treated as unnavigable anyway.

One thing to consider, is that in our current configuration, flag count has increased in some places fairly significantly because of the presence of navigable Ways within "Non-navigable" networks. Identifying these for fix I believe is a net positive to the map quality, not only from a navigability standpoint but also from an aesthetic one. Again, were this internal I would be more concerned with things such as editor time/fix count but because this is external and the main goal is map quality, I think this is the way to go. 

<img width="650" alt="Screen Shot 2019-06-26 at 11 19 10 AM" src="https://user-images.githubusercontent.com/37551535/60360951-c2432a00-9991-11e9-9849-4271a0ab162b.png">

Reply:
Thanks @kkonishi2 ! @danielduhh What is your thoughts on the above comment? Having a shared navigability criteria in atlas/atlas-checks would be ideal. I could work with @kkonishi2 to come up with validations for a shared car navigability function. Doesn't look like there is a unified navigability criteria in OSM as per @kkonishi2.

Reply:
Doesn't motor_vehicle/motorcar/vehicle=yes override access=no? The access tag is more general in terms of public access, whereas the transport mode tags are more car navigation specific

Reply:
thoughts?
@Bentleysb @nachtm

Reply:
From what I understand, access=no and motor_vehicle=yes means only motor_vehicles can drive down a road. So yes, a specific navigation tag should override access=no. This is supported by [the wiki page on the access tag](https://wiki.openstreetmap.org/wiki/Key:access): 

>  - access=no, bus=yes means that only buses are allowed to enter (for example a road only for buses)
> - access=yes, motor_vehicle=no means that all transport modes except motor vehicles can use the element
> - access=forestry, foot=permissive implies that forestry vehicles can use the route legally and that pedestrians can use it by currently but that permission may later be withdrawn.

Reply:
I feel like we need to decide on whether we should flag :
1) `all accessible highways that are not private -> access=!no and motorvehicle/motorcar/vehicle=!no` or
2) `public highways that are accessible -> access!=private and motorvehicle/motorcar/vehicle=yes`

Reply:
I have updated the conditions for car navigability to be access not in private enum set(as per the analysis done on SinkIsland) and transportation tags=yes. Feel like this should capture all the car navigable cases and skip all private roads.

Reply:
Micah's comment tells me that the access tag doesn't seem to be a reliable measure of actual accessibility. And I think a car navigable sink island is more problematic than a car non-navigable sink island because the former is impossible to enter/exit given their road network and despite access rights. So maybe we can flag ```motorvehicle/motorcar/vehicle=!no && isNotConnectedToCarNavigableEdge()``` and ignore the access tag.

Reply:
@keahi and I went through the different combinations and we feel like SinkIslandCheck should only look at edges that have access = yes and motorvehicle/vehicle/motorcar=yes tags. There are private roads with _motorvehicle/vehicle/motorcar tag value = yes_, like roads within a college campus, which may not be connected to any other navigable edges. These private roads need to be flagged as SinkIslands.